src/ui/marker.js

```diff
@@ -479,19 +487,22 @@ export default class Marker extends Evented {
     _calculateXYTransform(): string {
         const pos = this._pos;
         const map = this._map;
-
-        if (this.getPitchAlignment() !== 'map' || !map || !pos) { return ''; }
-        if (!map._usingGlobe()) {
-            const pitch = map.getPitch();
-            return pitch ? `rotateX(${pitch}deg)` : '';
+        const alignment = this.getPitchAlignment();
+
+        if (map && pos && alignment === 'map') {
+            if (!map._usingGlobe()) { // 'map' alignment on a flat map
+                const pitch = map.getPitch();
+                return pitch ? `rotateX(${pitch}deg)` : '';
+            } // "map" alignment on globe
+            const tilt = radToDeg(globeTiltAtLngLat(map.transform, this._lngLat));
+            const posFromCenter = pos.sub(globeCenterToScreenPoint(map.transform));
+            const tiltOverDist =  tilt / (Math.abs(posFromCenter.x) + Math.abs(posFromCenter.y));
+            const yTilt = posFromCenter.x * tiltOverDist;
+            const xTilt = -posFromCenter.y * tiltOverDist;
+            if (!xTilt && !yTilt) { return ''; }
```

Comment:
`!xTilt && !yTilt` are these condition checking against whether the variable is defined, or whether it's the value `0`? I assume it's against `0`? If so, we should prefer being being explicit

Reply:
Turns out this was both checking against 0 and `undefined` values created by the division by 0 as you mention in your previous comment. Updating the code to be more clear.